proj/vs2017/NAS2D.vcxproj

```diff
@@ -137,7 +137,7 @@
       <WarningLevel>Level4</WarningLevel>
       <Optimization>Disabled</Optimization>
       <SDLCheck>true</SDLCheck>
-      <AdditionalIncludeDirectories>%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <AdditionalIncludeDirectories>../../include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>

```

Comment:
This needs to be relative to a known macro path otherwise it will break if the project is ever moved.

I usually set it to $(SolutionDir): `$(SolutionDir)../../include;%(AdditionalIncludeDirectories)`

Changing while having "All Configurations" and "All Platforms" selected will also update all down-stream AID entries in this file.

Reply:
Relative directories should be fine. They will remain unchanged if the whole project is ever moved. I've used such paths in other projects quite extensively.

I'm a little uncertain if relative paths are relative to the project file or the solution file, though both are in the same folder, so would be the same.

----

You're right that we should probably look into removing some of the duplication in the project file at some point. Many settings are the same between all project configurations.

Reply:
They are relative to whatever you set it to:
 - Solution: `$(SolutionDir)`: "Append the directory path where the solution lives here"
 - Project: `$ProjectDir)`: "Append the directory path where the project lives here"

It just happens that they both live in the same place for this project. Usually the solution is at a higher directory level than a project, or a project is in an entirely different location.